threedi_modelchecker/checks/cross_section_definitions.py

```diff
@@ -323,63 +310,91 @@ def description(self):
-        if not heights:
-            heights = [0]
-        max_height = max(heights)
-        max_width = max(widths)
-        configuration = "closed"
-    elif shape == constants.CrossSectionShape.RECTANGLE.value:
-        if not widths:
-            widths = [0]
-        max_height = None
-        max_width = max(widths)
-        configuration = "open"
-    elif shape == constants.CrossSectionShape.CIRCLE.value:
-        if not widths:
-            widths = [0]
-        # any value filled in for heights will be overwritten by the widths value, also in the simulation
-        max_height = max_width = max(widths)
-        configuration = "closed"
+    if shape.is_tabulated:
+        raise ValueError("cross_section_configuration cannot handle tabulated shaptes")
+    max_width = 0 if not width else width
+    if shape == constants.CrossSectionShape.CLOSED_RECTANGLE:
+        max_height = 0 if not height else height
+    elif shape == constants.CrossSectionShape.RECTANGLE:
+        max_height = height
+    elif shape == constants.CrossSectionShape.CIRCLE:
+        max_height = max_width
     elif shape in [
-        constants.CrossSectionShape.EGG.value,
-        constants.CrossSectionShape.INVERTED_EGG.value,
+        constants.CrossSectionShape.EGG,
+        constants.CrossSectionShape.INVERTED_EGG,
     ]:
-        if not widths:
-            widths = [0]
-        # any value filled in for heights will be overwritten by 1.5 times the widths value, also in the simulation
-        max_width = max(widths)
         max_height = 1.5 * max_width
-        configuration = "closed"
-    elif shape in [
-        constants.CrossSectionShape.TABULATED_RECTANGLE.value,
-        constants.CrossSectionShape.TABULATED_TRAPEZIUM.value,
+    configuration = "closed" if shape.is_closed else "open"
+    return max_width, max_height, configuration
+
+
+def cross_section_configuration_tabulated(shape, widths, heights):
+    """
+    Retrieve maximum width, maximum height  and open/closed configuration for tabulated cross-sections.
+    """
+    if not shape.is_tabulated:
+        raise ValueError(
+            "cross_section_configuration_tabulated can only handle tabulated shaptes"
+        )
+    if not widths:
+        widths = [0]
+    if not heights:
+        heights = [0]
+    if shape in [
+        constants.CrossSectionShape.TABULATED_RECTANGLE,
+        constants.CrossSectionShape.TABULATED_TRAPEZIUM,
     ]:
-        if not widths:
-            widths = [0]
-        if not heights:
-            heights = [0]
         last_width = widths[-1]
         max_height = max(heights)
         max_width = max(widths)
         if last_width == 0:
             configuration = "closed"
         elif last_width > 0:
             configuration = "open"
-
-    elif shape == constants.CrossSectionShape.TABULATED_YZ.value:
-        if not widths:
-            widths = [0]
-        if not heights:
-            heights = [0]
+        else:
+            raise
```

Comment:
Not sure about this bare raise. IIRC Casper's policy was not to throw exceptions in the first place in the modelchecker, but otherwise perhaps a descriptive error message could be added to the exception to make it easier to debug when it throws an error in the QGIS plugin. Same applies to the raise on line 410.